lib/ws_servers/api/algos/algo_worker.js

```diff
@@ -296,6 +297,32 @@ class AlgoWorker {
     await host.stopAO(gid)
   }
 
+  /**
+   * @public
+   * @param {string} gid
+   * @param {number} lastActive
+   * @returns {Promise<void>}
+   */
+  async updateLastActive (gid, lastActive) {
+    if (!this.isHostAvailable()) {
+      return Promise.reject(new Error('Internal error, host is not available'))
+    }
+
+    const { host } = this
+
+    const instance = host.getAOInstance(gid)
+    if (!instance) {
+      const error = new Error('Requested algo order not running, cannot stop')
+
+      error.i18n = applyI18N(['requestedAlgoOrderNotRunningCannotStop'])
+      throw error
+    }
+
+    const serialized = host.getSerializedAO(instance)
```

Comment:
this code is duplicated, can we extract that to a `this._getSerializedAO` method and change `cancelOrder` too?

Reply:
good point